### PR TITLE
Add schema dumping support for PostgreSQL geometric data types

### PR DESCRIPTION
**Pull request by kamipo**

### CODE REVIEW

File: activerecord/test/cases/adapters/postgresql/geometric_test.rb [hunk 1]
```diff
@@ -212,10 +217,12 @@ def test_alternative_format
       :a_polygon      => '2.0, 3, 5.5, 7.0, 8.5, 11.0',
       :a_circle       => '((5.3, 10.4), 2)'
     )
+    g[:a_line] = '(2.0, 3), (4.0, 6.0)' if supports_line?
 
     g.save!
 
     h = PostgresqlGeometric.find(g.id)
+    assert_equal '{1.5,-1,0}', h.a_line if supports_line?
```
Comment (senny) [inline activerecord/test/cases/adapters/postgresql/geometric_test.rb]: Can we put stuff about the `line` type in a separate test? I don't like how `supports_line?` is spread all over this test.

**Comment by senny** [comment]
@kamipo just a minor note. Otherwise this is looking good. Can we close #22188 in favor of this PR?

**Comment by kamipo** [comment]
Fixed the `line` type test in a separate test. I'll close #22188, thanks!

**Comment by senny** [comment]
@kamipo this looks good, can you push a rebased version?

File: activerecord/test/cases/adapters/postgresql/geometric_test.rb [hunk 2]
```diff
@@ -167,16 +167,19 @@ def test_legacy_mutation
...
 
   setup do
     @connection = ActiveRecord::Base.connection
     @connection.create_table("postgresql_geometrics") do |t|
-      t.column :a_line_segment, :lseg
-      t.column :a_box, :box
-      t.column :a_path, :path
-      t.column :a_polygon, :polygon
-      t.column :a_circle, :circle
+      t.line    :a_line
```
Comment (senny) [inline activerecord/test/cases/adapters/postgresql/geometric_test.rb]: Does this `line` type not follow the same restrictions for PostgreSQL version? I think we should check the dump in `PostgreSQLGeometricLineTest` as well and remove that from here.

**Comment by kamipo** [comment]
I moved all `line` type tests moved to `PostgreSQLGeometricLineTest`.

**Comment by senny** [comment]
@kamipo thank you :yellow_heart:

**Pull request merged**